src/pcre2_compile.c

```diff
@@ -11022,17 +11048,17 @@ there are no occurrences of *PRUNE or *SKIP (though there is an option to
 disable this case). */
 
 if ((re->overall_options & PCRE2_ANCHORED) == 0 &&
-     is_anchored(codestart, 0, &cb, 0, FALSE))
+     is_anchored(codestart, 0, &cb, 0, FALSE, optim_flags & PCRE2_OPTIM_DOTSTAR_ANCHOR))
```

Comment:
nitpick: since the parameter is a BOOL then a uint32_t needs to be compared with something here.
I know it is unusual and not needed for C but that is just the way it is done in this code base.  That is also why most pointers are explicitly compared to NULL FWIW.

Reply:
I did a lot of my early computing on a system where user memory addresses began at zero. If C had been invented at that time, NULL could not have been zero in that environment. That is one reason why I always compare pointers to NULL explicitly. The other is that it makes it clear that the variable is a pointer. That's also why I write "if (x)" only when x is a Boolean; if it's a numerical variable I write "if (x == 0)" to make that clear. Call me a pedant.....

Reply:
I also prefer comparing with NULL, since it shows that a pointer result is present. It is easier to understand the code later (or by somebody else).

Reply:
As Philip pointed out there are some systems out there with a non zero NULL so that is the "correct" thing to do, at least for pointers. I suspect though in more systems were PCRE2 runs nowadays there is no difference.

Reply:
Good points, all.